core/modules/replica/WorkerIndexRequest.cc

```diff
@@ -52,30 +52,44 @@ namespace lsst {
 namespace qserv {
 namespace replica {
 
-WorkerIndexRequest::Ptr WorkerIndexRequest::create(ServiceProvider::Ptr const& serviceProvider,
-                                                   ConnectionPoolPtr const& connectionPool,
-                                                   string const& worker,
-                                                   string const& id,
-                                                   ProtocolRequestIndex const& request) {
-    return WorkerIndexRequest::Ptr(
-        new WorkerIndexRequest(serviceProvider,
-                               connectionPool,
-                               worker,
-                               id,
-                               request));
+WorkerIndexRequest::Ptr WorkerIndexRequest::create(
+        ServiceProvider::Ptr const& serviceProvider,
+        ConnectionPoolPtr const& connectionPool,
+        string const& worker,
+        string const& id,
+        int priority,
+        ExpirationCallbackType const& onExpired,
+        unsigned int requestExpirationIvalSec,
+        ProtocolRequestIndex const& request) {
+    return WorkerIndexRequest::Ptr(new WorkerIndexRequest(serviceProvider,
+        connectionPool,
+        worker,
+        id,
+        priority,
+        onExpired,
+        requestExpirationIvalSec,
+        request
+    ));
```

Comment:
There are a lot of these (half dozen arguments passed from create to constructor to constructor again). Would it help to have a data classes for WorkerRequest and all of its children? Then you could define WorkerRequestData and WorkerIndexRequestData and use them as arguments to the create functions and constructor functions. Reduced numbers of arguments passed and reduced duplicate comments.
``` 
struct WorkerRequestData {
    //  copy constructor
    std::string const worker;
    std::string const type;
    std::string const id;
    int         const priority;
    ExpirationCallbackType onExpired;
};

class WorkerRequest : public std::enable_shared_from_this<WorkerRequest> {
public:
    WorkerReques(WorkerRequestData const& workerRequestData) : _data(workerRequestData) { ... }
 ...
private:
   WorkerRequestData _data;
...
};
```

Reply:
That's an interesting suggestion! I will investigate it

Reply:
My experience with such "data" classes is not good; they can introduce ownership confusion and when refactoring you may have to modify things in unexpected locations. I think the problem John sites is a symptom of too-much-boilerplate, but IMHO the proposed fix is an anti pattern.

Reply:
Okay, since I have to opposite opinions on this subject then I choose do nothing 😸